WordPress/Classes/Services/PostService+Revisions.swift

```diff
@@ -29,6 +29,26 @@ extension PostService {
         }, failure: failure)
     }
 
+    /// Check if the given post matches the latest revision on the site and logs an error event if it doesn't.
+    @objc(checkLatestRevisionForPost:usingRemote:)
+    func checkLatestRevision(for post: AbstractPost, using remote: PostServiceRemote) -> Void {
+        guard let postID = post.postID,
+              postID.int64Value > 0,
+              let restApi = (remote as? PostServiceRemoteREST),
+              let localRevision = post.revisions?.first as? NSNumber
+        else {
+            return
+        }
+
+        restApi.getPostLatestRevisionID(for: postID) { latestRemoteRevision in
+            if let latestRemoteRevision, latestRemoteRevision != localRevision {
+                DDLogError("The latest revision (\(latestRemoteRevision)) of the post (id: \(postID) is about to be overwriten by an edit that's based on revision \(localRevision)")
```

Comment:
Nitpick. This log assumes the method is called before `post` is about to be saved, but that's not necessarily the case. 

Have you considered delegating the handling of what to do based on the comparison result?
 
```suggestion
                DDLogError("The latest revision (\(latestRemoteRevision)) of the post (id: \(postID)) is about to be overwritten by an edit that's based on revision \(localRevision)")
```

Reply:
> based on the comparison result?

😮 this made me think of returning something like [`NSComparisonResult`](https://developer.apple.com/documentation/foundation/nscomparisonresult), but it might be best to spend energy in addressing the root of the issue rather than gold-plating making the diagnostic